### PR TITLE
SSBFFA: Ban Certain Abilities

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: spacialgaze-plugins/ssb.js [hunk 1]
```diff
@@ -388,6 +388,7 @@ class SSB {
 			for (let i in Dex.getTemplate(this.species).abilities) {
 				if (toId(Dex.getTemplate(this.species).abilities[i]) === ability.id) {
 					this.ability = ability.name;
+					if (ability.id === 'powerconstruct') return false;
```
Comment (DeathlyPlays) [inline spacialgaze-plugins/ssb.js]: Ban Arena Trap also

Comment (ghost) [reply spacialgaze-plugins/ssb.js]: K

Comment (ghost) [reply spacialgaze-plugins/ssb.js]: k

**Comment by ghost** [comment]
Bah i give up on this. >_>

File: spacialgaze-plugins/ssb.js [hunk 2]
```diff
@@ -466,7 +471,7 @@ class SSB {
 		return true;
 	}
 	activate() {
-		if (this.species && (this.movepool.length > 0 || this.cMove) && this.ability) {
+		if (this.species && (this.movepool.length > 0 || this.cMove) && (this.ability && ability.id !== 'shadowtag')) {
```
Comment (DeathlyPlays) [inline spacialgaze-plugins/ssb.js]: Maybe use || instead of && (and add the other banned abilities)

**Review by DeathlyPlays — COMMENTED** [review]
Dude, you're almost done lmfao

File: spacialgaze-plugins/ssb.js (hunk 2 above)
Comment (HoeenCoder) [reply spacialgaze-plugins/ssb.js]: Why are you adding a validation check to `activate()` anyways? True validation is handled elsewhere (idr the exact function name, but its called by `/ssb validate`) The validation here is just to make sure the pokemon has the base information needed to not crash the server when its selected to battle.

File: spacialgaze-plugins/ssb.js [hunk 3]
```diff
@@ -816,7 +821,7 @@ exports.commands = {
 					return this.sendReply('Your pokemon was deactivated. Your pokemon will no longer appear in battles once the server restarts.');
 				}
 			} else {
-				return this.errorReply('Could not activate your pokemon, all pokemon must have at least 1 move.');
+				return this.errorReply('Could not activate your pokemon, all pokemon must have at least 1 move or you have an illegal ability.');
```
Comment (HoeenCoder) [inline spacialgaze-plugins/ssb.js]: Also unnessecary

File: spacialgaze-plugins/ssb.js (hunk 2 above)
Comment (ghost) [reply spacialgaze-plugins/ssb.js]: I know i shouldn’t >_>

It’s just wobbuffet was being stubborn

File: spacialgaze-plugins/ssb.js (hunk 3 above)
Comment (ghost) [reply spacialgaze-plugins/ssb.js]: Oh ok

**Review by ghost — COMMENTED** [review]
In line 289 Seems the best place to put the check...

**Review by DeathlyPlays — APPROVED** [review]

File: spacialgaze-plugins/ssb.js [hunk 4]
```diff
@@ -286,7 +286,22 @@ class SSB {
...
+			} else if (species.abilities['1'] !== 'powerconstruct' || species.abilities['1'] !== 'arenatrap' || species.abilities['1'] !== 'shadowtag') {
+				this.ability = species.abilities['1']; //Force legal ability
+				break;
+			} else if (species.abilities['H'] !== 'powerconstruct' || species.abilities['H'] !== 'arenatrap' || species.abilities['H'] !== 'shadowtag') {
+				this.ability = species.abilities['H']; //Force legal ability
+				break;
+			} else {
+				this.ability = species.abilities['0']; //Force legal ability
+				return this.errorReply('An error occurred. Defaulting to first ability.'); //This shouldn't happen
+				break;
```
Comment (DeathlyPlays) [inline spacialgaze-plugins/ssb.js]: Don't use break and return

Comment (ghost) [reply spacialgaze-plugins/ssb.js]: ah

File: spacialgaze-plugins/ssb.js [hunk 5]
```diff
@@ -287,9 +287,9 @@ class SSB {
 		}
 		this.species = species.species;
 		//Ability
-		let AbilityCheck = true
-		if (AbilityCheck = true) {
-			while (AbilityCheck = true) /*This is really hacky but hopefully does the trick*/ {
+		let AbilityCheck === true; /* eslint  no-unused-vars: [0] */
```
Comment (DeathlyPlays) [inline spacialgaze-plugins/ssb.js]: Shouldn't need three ``=`` since its just defining a variable

Comment (DeathlyPlays) [reply spacialgaze-plugins/ssb.js]: May be wrong

**Comment by ghost** [comment]
alright it passed travis let’s see what the actual test phase results in

File: spacialgaze-plugins/ssb.js [hunk 6]
```diff
@@ -288,8 +288,8 @@ class SSB {
 		this.species = species.species;
 		//Ability
 		let AbilityCheck = true; /* eslint  no-unused-vars: [0] */
-		if (AbilityCheck === true) {
-			while (AbilityCheck === true) /*This is really hacky but hopefully does the trick*/ {
+		if (AbilityCheck = true) {
```
Comment (HoeenCoder) [inline spacialgaze-plugins/ssb.js]: Why are you setting it to true in an if statement?

File: spacialgaze-plugins/ssb.js [hunk 7]
```diff
@@ -288,8 +288,8 @@ class SSB {
...
+			while (AbilityCheck = true) /*This is really hacky but hopefully does the trick*/ {
```
Comment (HoeenCoder) [inline spacialgaze-plugins/ssb.js]: Infinite loop!!!

**Comment by ghost** [comment]
Im gonna start this over, i feel like there should be a better way and this is just to big to try and edit it

**Comment by ghost** [comment]
Also this PR is lagging.


its for the best of githib that i continue this somewhere else

**Comment by HoeenCoder** [comment]
ok

**Comment by DeathlyPlays** [comment]
Fair enough

**Pull request closed without merging**